react/features/local-recording/components/LocalRecordingInfoDialog.js

```diff
@@ -0,0 +1,332 @@
+import { connect } from 'react-redux';
+
+import { translate } from '../../base/i18n';
+import {
+    PARTICIPANT_ROLE,
+    getLocalParticipant
+} from '../../base/participants';
+
+import { statsUpdate } from '../actions';
+import { recordingController } from '../controller';
+
+
+/**
+ * The type of the React {@code Component} props of
+ * {@link LocalRecordingInfoDialog}.
+ */
+type Props = {
+
+    /**
+     * Redux store dispatch function.
+     */
+    dispatch: Dispatch<*>,
+
+    /**
+     * Current encoding format.
+     */
+    encodingFormat: string,
+
+    /**
+     * Whether the local user is the moderator.
+     */
+    isModerator: boolean,
+
+    /**
+     * Whether local recording is engaged.
+     */
+    isOn: boolean,
+
+    /**
+     * The start time of the current local recording session.
+     * Used to calculate the duration of recording.
+     */
+    recordingStartedAt: Date,
+
+    /**
+     * Stats of all the participant.
+     */
+    stats: Object,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
+    t: Function
+}
+
+/**
+ * The type of the React {@code Component} state of
+ * {@link LocalRecordingInfoDialog}.
+ */
+type State = {
+
+    /**
+     * The recording duration string to be displayed on the UI.
+     */
+    durationString: string
+}
+
+/**
+ * A React Component with the contents for a dialog that shows information about
+ * local recording. For users with moderator rights, this is also the "control
+ * panel" for starting/stopping local recording on all clients.
+ *
+ * @extends Component
+ */
+class LocalRecordingInfoDialog extends Component<Props, State> {
+
+    /**
+     * Saves a handle to the timer for UI updates,
+     * so that it can be cancelled when the component unmounts.
+     */
+    _timer: ?IntervalID;
+
+    /**
+     * Constructor.
+     */
+    constructor() {
+        super();
+        this.state = {
+            durationString: 'N/A'
+        };
+    }
+
+    /**
+     * Implements React's {@link Component#componentWillMount()}.
+     *
+     * @returns {void}
+     */
+    componentWillMount() {
+        this._timer = setInterval(
+            () => {
+                this.setState((_prevState, props) => {
+                    const nowTime = new Date(Date.now());
+
+                    return {
+                        durationString: this._getDuration(nowTime,
+                            props.recordingStartedAt)
+                    };
+                });
+                try {
+                    this.props.dispatch(
+                        statsUpdate(recordingController
+                            .getParticipantsStats()));
+                } catch (e) {
+                    // do nothing
+                }
+            },
+            1000
+        );
+    }
+
+    /**
+     * Implements React's {@link Component#componentWillUnmount()}.
+     *
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        if (this._timer) {
```

Comment:
You are right. But flow will complain if I remove the null check: 
```
Cannot call `clearInterval` with `this._timer` bound to `intervalId` because null or undefined [1] is incompatible with
`IntervalID` [2].
```